include/flang/ISO_Fortran_binding.h

```diff
@@ -122,13 +130,17 @@ typedef struct CFI_cdesc_t {
   CFI_type_t type;
   CFI_attribute_t attribute;
   unsigned char f18Addendum;
-  CFI_dim_t dim[CFI_ISO_FORTRAN_BINDING_FLEXIBLE_ARRAY]; /* must appear last */
+#ifdef __cplusplus
+  CFI_dim_t_cpp_flexible_array dim;
```

Comment:
or templatize it as `FlexibleArray<CFI_dim_t>`

Reply:
Looks nicer, thanks. Addressed in 7b1fc24. I also added the cfi_internal namespace around to avoid spilling definitions in user program.